com.unity.ml-agents/Runtime/Inference/BarracudaModelExtensions.cs

```diff
@@ -298,21 +312,33 @@ public static bool SupportsContinuousAndDiscrete(this Model model)
         /// <param name="failedModelChecks">Output list of failure messages</param>
         ///
         /// <returns>True if the model contains all the expected tensors.</returns>
-        public static bool CheckExpectedTensors(this Model model, List<string> failedModelChecks)
+        public static bool CheckExpectedTensors(this Model model, List<FailedCheck> failedModelChecks)
```

Comment:
Not directly from this PR, but this might be cleaner with a `out FailedCheck failedModelCheck` parameter instead, since we're never pushing more than one.

Reply:
I tried to play around, but  do not think it ill be cleaner, because when this method is called, there is already a list of failed checks that is being constructed.

Reply:
OK, not a big deal. The whole thing might be a little cleaner by `yield`ing errors, but that's a bigger change too.